src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -67,11 +67,12 @@ public void Initialize(EntityInfo entity, Func<IncomingMessageDetails, ReceiveCo
 
             numberOfClients = settings.Get<int>(WellKnownConfigurationKeys.Connectivity.NumberOfClientsPerEntity);
             var concurrency = maximumConcurrency / (double)numberOfClients;
+            var maxConcurrentCalls = concurrency > 1 ? (int) Math.Round(concurrency, MidpointRounding.AwayFromZero) : 1;
```

Comment:
Example: concurrency is set to 2 and number of clients to 10. It will cause concurrency to be lifted to 10. I think the least we could do to address it is output a warning message explaining what happened.